main-svc/src/main/java/ru/practicum/explorewithme/mainsvc/location/service/LocationServiceImpl.java

```diff
@@ -2,51 +2,184 @@
+    @Override
+    public LocationDto addLocation(LocationDto locationDto) {
+        if (locationRepository.existsByLatAndLonAndRadiusAndNameAndVerified(
+                locationDto.getLat(), locationDto.getLon(), locationDto.getRadius(), locationDto.getName(), true)
+        ) {
+            throw new AlreadyExistsException("Verified location " + locationDto + " already exists.");
+        }
+        Location location = locationMapper.toEntity(locationDto);
+
+        if (location.getRadius() == null) {
+            location.setRadius(0.01);
+        }
+        location.setVerified(true);
+        Location savedLocation = locationRepository.save(location);
+
+        LocationDto result = locationMapper.toDto(savedLocation);
+        log.info("Location has been created : {}.", result);
+        return result;
+    }
+
+    @Transactional
+    @Override
+    public LocationDto updateLocation(long id, LocationUpdateDto locationDto) {
+        Location location = findLocationById(id);
+
+        updateLocationProperties(location, locationDto);
+        Location savedLocation = locationRepository.save(location);
+
+        LocationDto result = locationMapper.toDto(savedLocation);
+        log.info("Location has been updated : {}.", result);
+        return result;
+    }
+
+    @Transactional
+    @Override
+    public void deleteLocations(List<Long> ids) {
+        if (!locationRepository.existsAllByIdIn(ids)) {
+            throw new EntityNotFoundException("Not all locations by ids " + ids + " not found.");
+        }
+        if (eventRepository.existsByLocationIdIn(ids)) {
+            throw new AccessRightsException("There are events at locations with ids " + ids + ".");
+        }
+        locationRepository.deleteAllByIdIn(ids);
+        log.info("Locations {} has been deleted.", ids);
+    }
 
     @Transactional(readOnly = true)
     @Override
-    public boolean isLocationExists(Location location) {
-        return locationRepository.existsById(LocationPrimaryKey.builder()
-                .lat(location.getLat())
-                .lon(location.getLon())
-                .build()
-        );
+    public LocationDto getLocation(long id) {
+        Location location = findLocationById(id);
+        LocationDto result = locationMapper.toDto(location);
+        log.info("Location has been found : {}.", result);
+        return result;
+    }
+
+    @Transactional(readOnly = true)
+    @Override
+    public List<LocationDto> getLocationsByAdmin(LocationsAdminRequest locationsRequest,
+                                                 PaginationRequest paginationRequest) {
+        Pageable pageable = pageableUtility.toPageable(paginationRequest);
+
+        List<Location> locations;
+        if (Boolean.TRUE.equals(locationsRequest.getVerified())) {
+            locations = locationRepository.findByVerified(true, pageable);
+        } else {
+            locations = locationRepository.findByVerified(false, pageable);
+        }
```

Comment:
```suggestion
            locations = locationRepository.findByVerified(locationsRequest.getVerified(), pageable);
```
не нужно усложнять логику